src/main/java/com/petproject/pomodoro/service/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.petproject.pomodoro.service;
+
+
+import com.petproject.pomodoro.config.UserPrincipal;
+import com.petproject.pomodoro.entity.User;
+import com.petproject.pomodoro.exceptions.NoSuchElementException;
+import com.petproject.pomodoro.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private UserRepository userRepository;
+
+    @Autowired
+    public UserDetailsServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```

Comment:
about "throws UsernameNotFoundException":
1) Do we throw this exception?
2) Should we define runtime exception for this method if we do not define them for all other methods, where we can throw them?

Reply:
1. One way to do it properly, as see it, is to create custom exception, make it extend `UsernameNotFoundException ` (could be named `UserNotFoundException` for example)  and to implement Supplier interface, cause that is what `.orElseThrow` requires.

2. I'm not sure yet but I think we would be using `UserNotFoundException` in the future somewhere else.

Reply:
Ok, I liked the first option. 
For example, we could rename NoSuchElementException, which is currently used only by users into UserNotFoundException (which is also a good option because it will not coincide with a default name like NoSuchElementException). And then extend from UsernameNotFoundException, in order to throw something that is expected by the UserDetailsService interface.
And after that, we may remove “throws UsernameNotFoundException”, because it is unnecessary information (we already could see this exception in the method's body + we do not usually include runtime exceptions into methods signature).